src/prlib/prlib.cpp

```diff
@@ -68,91 +91,162 @@ INCLUDE_ASM(const s32, "prlib/prlib", PrCleanupCamera);
+}
 
-INCLUDE_ASM(const s32, "prlib/prlib", PrSetCameraUserData);
+PR_EXTERN void PrSetCameraUserData(PR_CAMERAHANDLE camera, int userData)
+{
+    ((SpcFileHeader*)camera)->m_userData = userData;
+}
 
-INCLUDE_ASM(const s32, "prlib/prlib", PrGetModelUserData);
+PR_EXTERN int PrGetModelUserData(PR_MODELHANDLE model)
+{
+    return ((PrModelObject*)model)->m_userData;
+}
 
-INCLUDE_ASM(const s32, "prlib/prlib", PrGetAnimationUserData);
+PR_EXTERN int PrGetAnimationUserData(PR_ANIMATIONHANDLE animation)
+{
+    return ((SpaFileHeader*)animation)->m_userData;
+}
 
-INCLUDE_ASM(const s32, "prlib/prlib", PrGetCameraUserData);
+PR_EXTERN int PrGetCameraUserData(PR_CAMERAHANDLE camera)
+{
+    return ((SpcFileHeader*)camera)->m_userData;
+}
 
-INCLUDE_ASM(const s32, "prlib/prlib", PrLinkAnimation);
+PR_EXTERN void PrLinkAnimation(PR_MODELHANDLE model, PR_ANIMATIONHANDLE animation)
+{
+    ((PrModelObject*)model)->LinkAnimation((SpaFileHeader*)animation);
+}
 
-INCLUDE_ASM(const s32, "prlib/prlib", PrUnlinkAnimation);
+PR_EXTERN void PrUnlinkAnimation(PR_MODELHANDLE model)
+{
+    ((PrModelObject*)model)->LinkAnimation(NULL);
+}
 
-INCLUDE_ASM(const s32, "prlib/prlib", PrGetLinkedAnimation);
+PR_EXTERN PR_ANIMATIONHANDLE PrGetLinkedAnimation(PR_MODELHANDLE model)
+{
+    return (PR_ANIMATIONHANDLE)((PrModelObject*)model)->m_linkedAnim;
+}
 
-INCLUDE_ASM(const s32, "prlib/prlib", PrLinkPositionAnimation);
+PR_EXTERN void PrLinkPositionAnimation(PR_MODELHANDLE model, PR_ANIMATIONHANDLE animation)
+{
+    ((PrModelObject*)model)->LinkPositionAnimation((SpaFileHeader*)animation);
+}
 
-INCLUDE_ASM(const s32, "prlib/prlib", PrUnlinkPositionAnimation);
+PR_EXTERN void PrUnlinkPositionAnimation(PR_MODELHANDLE model)
+{
+    ((PrModelObject*)model)->LinkPositionAnimation(NULL);
+}
 
-INCLUDE_ASM(const s32, "prlib/prlib", PrGetLinkedPositionAnimation);
+PR_EXTERN PR_ANIMATIONHANDLE PrGetLinkedPositionAnimation(PR_MODELHANDLE model)
+{
+    return (PR_ANIMATIONHANDLE)((PrModelObject*)model)->m_linkedPosAnim;
+}
 
-INCLUDE_ASM(const s32, "prlib/prlib", PrSelectCamera);
+PR_EXTERN void PrSelectCamera(PR_CAMERAHANDLE camera, PR_SCENEHANDLE scene)
+{
+    ((PrSceneObject*)scene)->SelectCamera((SpcFileHeader*)camera);
+}
 
-INCLUDE_ASM(const s32, "prlib/prlib", PrGetSelectedCamera);
+PR_EXTERN PR_CAMERAHANDLE PrGetSelectedCamera(PR_SCENEHANDLE scene)
+{
+    return (PR_CAMERAHANDLE)((PrSceneObject*)scene)->m_pCurrentCamera;
+}
 
-INCLUDE_ASM(const s32, "prlib/prlib", PrGetCurrentCamera);
+PR_EXTERN void* PrGetCurrentCamera(PR_SCENEHANDLE scene)
+{
+    return ((PrSceneObject*)scene)->GetCurrentCamera();
+}
 
 INCLUDE_ASM(const s32, "prlib/prlib", PrSetDefaultCamera);
 
 INCLUDE_ASM(const s32, "prlib/prlib", PrSetAppropriateDefaultCamera);
 
 INCLUDE_ASM(const s32, "prlib/prlib", PrShowModel);
 
-PR_EXTERN NaMATRIX<float, 4, 4>* PrGetModelMatrix(PrModelObject *model)
+PR_EXTERN float* PrGetModelMatrix(PR_MODELHANDLE model)
```

Comment:
Don't remove NaMATRIX - same on the other functions.

Reply:
NaMATRIX and NaVECTOR are C++ structs and are not compatible with the C definitions as far as I know